File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/prototype/transport/Plane.java
// Plane.java

package com.bigboxcode.designpattern.prototype.transport;

public class Plane implements Prototype {
    public String model;
    public String color;

    public Plane() {

    }

    public Plane(Plane plane) {
        this.model = plane.model;
        this.color = plane.color;
    }

    @Override
    public Prototype clone() {
        return new Plane(this);
    }

    public String toString() {
        return "Model: " + model + " | Color: " + color;
    }
}

File: design-patterns/php/factory/transportmethod/Plane.php
<?php
// Plane.php

namespace BigBoxCode\DesignPattern\Factory\TransportMethod;

class Plane implements Transport {
    public function start() {
        echo "Plane started\n";
    }

    public function stop() {
        echo "Plane Stopped\n";
    }

    public function repair() {
        echo "Plane Repair\n";
    }
}